add warnings to L1TGlobalPrescaler (93x)

fwyzard opened the pull request:
- warn if `l1tPrescales` has the wrong number of elements
  - warn if the target prescales are lower than the original ones

fwyzard committed:
add warnings to L1TGlobalPrescaler

  - warn if `l1tPrescales` has the wrong number of elements
  - warn if the target prescales are lower than the original ones

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @fwyzard (Andrea Bocci) for master.

It involves the following packages:

L1Trigger/L1TGlobal


@cmsbuild, @rekovic, @mulhearn can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @kreczko this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

fwyzard commented:
bugfix

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-20144/76

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-20144/76/git-diff.patch

In future, you can run `scram build code-checks` to apply code checks

fwyzard committed:
cleaned up by clang-tidy 4.0.0 modernize checks

Enabled checks:
    boost-use-to-string
    cppcoreguidelines-no-malloc
    google-build-explicit-make-pair
    misc-inefficient-algorithm
    misc-string-compare
    misc-uniqueptr-reset-release
    modernize-avoid-bind
    modernize-deprecated-headers
    modernize-loop-convert
    modernize-make-shared
    modernize-make-unique
    modernize-pass-by-value
    modernize-redundant-void-arg
    modernize-replace-auto-ptr
    modernize-shrink-to-fit
    modernize-use-auto
    modernize-use-bool-literals
    modernize-use-emplace
    modernize-use-equals-default
    modernize-use-equals-delete
    modernize-use-nullptr
    modernize-use-override
    modernize-use-transparent-functors
    modernize-use-using
    performance-faster-string-find
    performance-for-range-copy
    performance-implicit-cast-in-loop
    performance-inefficient-string-concatenation
    performance-type-promotion-in-math-fn
    performance-unnecessary-copy-initialization
    performance-unnecessary-value-param
    readability-container-size-empty
    readability-redundant-string-cstr
    readability-static-definition-in-anonymous-namespace
    readability-uniqueptr-delete-release

See http://releases.llvm.org/4.0.0/tools/clang/tools/extra/docs/clang-tidy/index.html for details.

cmsbuild commented:
Pull request #20144 was updated. @cmsbuild, @rekovic, @mulhearn can you please check and sign again.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pr-code-checks/PR-20144/77

rekovic commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
+1
Tested at: 6113c96c653a72a6ebecd004c74008bc62f1da20
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20144/22252/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/56f2f723d8eee5e34bf7ce5ac2e6871fdfa7886d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20144/22252/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20144/22252/git-merge-result

fwyzard commented on L1Trigger/L1TGlobal/plugins/L1TGlobalPrescaler.cc:
```diff
@@ -100,18 +99,38 @@ namespace {
+
+namespace {
+
+  template <typename T, std::size_t N, typename S>
```
@Dr15Jones @wddgit can you think of any way to support
```c++
template <typename T, std::size_t N>
std::array<T, N>
ParameterSet::getParameter<std::array<T, N>>(...)
```
directly in `ParameterSet` ?

I I understand that it cannot be done as an overload or partial specialisation of `ParameterSet::getParameter<T>(...)` .

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20144/22252/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2600817
 - DQMHistoTests: Total failures: 240
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2600398
 - DQMHistoTests: Total skipped: 179
 - DQMHistoTests: Total Missing objects: 0
 - Checked 103 log files, 14 edm output root files, 25 DQM output files

davidlange6 commented:
+1
